src/mmw/apps/modeling/mapshed/tasks.py

```diff
@@ -0,0 +1,171 @@
+    data['input'].update(input_data)
+
+    job_id = sjs_submit(host, port, args, data)
+
+    return {
+        'host': host,
+        'port': port,
+        'job_id': job_id
+    }
+
+
+@shared_task(bind=True, default_retry_delay=1, max_retries=42)
+def mapshed_finish(self, incoming):
+    sjs_result = sjs_retrieve(retry=self.retry, **incoming)
+
+    # Convert string "List(1,2,3)" into tuple (1,2,3) for each key
+    return {make_tuple(key[4:]): val for key, val in sjs_result.items()}
+
+
+@shared_task
+def collect_data(geop_results, geojson):
+    geom = GEOSGeometry(geojson, srid=4326)
+    area = geom.transform(5070, clone=True).area  # Square Meters
+
+    # Data Model is called z by convention
+    z = DataModel(settings.GWLFE_DEFAULTS)
+
+    # Statically calculated lookup values
+    z.DayHrs = day_lengths(geom)
+
+    # Data from the Weather Stations dataset
+    ws = nearest_weather_stations(geom)
+    z.Grow = growing_season(ws)
+    z.Acoef = erosion_coeff(ws, z.Grow)
+    z.PcntET = et_adjustment(ws)
+    z.WxYrBeg = max([w.begyear for w in ws])
+    z.WxYrEnd = min([w.endyear for w in ws])
+    z.WxYrs = z.WxYrEnd - z.WxYrBeg + 1
+
+    # Data from the County Animals dataset
+    livestock_aeu, poultry_aeu = animal_energy_units(geom)
+    z.AEU = livestock_aeu / (area * ACRES_PER_SQM)
+    z.n41j = livestock_aeu
+    z.n41k = poultry_aeu
+    z.n41l = livestock_aeu + poultry_aeu
+
+    z.ManNitr, z.ManPhos = manure_spread(z.AEU)
+
+    # Data from Streams dataset
+    z.StreamLength = stream_length(geom)   # Meters
+    z.n42b = round(z.StreamLength / 1000)  # Kilometers
+
+    # Data from Point Source Discharge dataset
+    n_load, p_load, discharge = point_source_discharge(geom, area)
+    z.PointNitr = n_load
+    z.PointPhos = p_load
+    z.PointFlow = discharge
+
+    # Data from National Weather dataset
+    temps, prcps = weather_data(ws, z.WxYrBeg, z.WxYrEnd)
+    z.Temp = temps
+    z.Prec = prcps
+
+    # Flatten geoprocessing results into one dictionary
+    geop_result = {k: v for r in geop_results for k, v in r.items()}
+
+    z.AgLength = geop_result['AgStreamPct'] * z.StreamLength
+    z.UrbLength = z.StreamLength - z.AgLength
+
+    # TODO pass real input to model instead of reading it from gms file
+    gms_filename = join(dirname(abspath(__file__)), 'data/sample_input.gms')
+    gms_file = open(gms_filename, 'r')
+    z = parser.GmsReader(gms_file).read()
+
+    # The frontend expects an object with runoff and quality as keys.
+    response_json = {'runoff': gwlfe.run(z)}
+
+    return response_json
+
+
+@shared_task
+def start_gwlfe_job(model_input):
+    geom = GEOSGeometry(json.dumps(model_input['area_of_interest']),
+                        srid=4326)
+
+    return (group(geop_tasks(geom)) |
+            identity.s() |
+            collect_data.s(geom.geojson))
+
+
+@shared_task
+def nlcd_streams(result):
+    ag_streams = sum(result.get(nlcd, 0) for nlcd in [81, 82])
```

Comment:
Alternatively:

This might be better to do in the next step, but assuming each of the new functions you'll need to write for the next feature have similar structure, would it be worthwhile to make a sort of wrapper function that takes in `result`, some functions for manipulating it in whatever way, and then the key-val pair to return as an object and uses them all to generate the result? 

Then you could document how to use the wrapper fn rather than documenting how to mimic this one to generate other values.